pkg/ccl/crosscluster/physical/stream_ingestion_job_test.go

```diff
@@ -298,6 +290,113 @@ func TestTenantStreamingFailback(t *testing.T) {
+
+	skip.UnderRace(t, "test takes ~5 minutes under race")
+
+	serverA, aDB, _ := serverutils.StartServer(t, base.TestServerArgs{
+		DefaultTestTenant: base.TestControlsTenantsExplicitly,
+		Knobs: base.TestingKnobs{
+			JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
+		},
+	})
+	defer serverA.Stopper().Stop(ctx)
+	serverB, bDB, _ := serverutils.StartServer(t, base.TestServerArgs{
+		DefaultTestTenant: base.TestControlsTenantsExplicitly,
+		Knobs: base.TestingKnobs{
+			JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
+		},
+	})
+	defer serverB.Stopper().Stop(ctx)
+
+	sqlA := sqlutils.MakeSQLRunner(aDB)
+	sqlB := sqlutils.MakeSQLRunner(bDB)
+
+	serverAURL, cleanupURLA := sqlutils.PGUrl(t, serverA.SQLAddr(), t.Name(), url.User(username.RootUser))
+	defer cleanupURLA()
+
+	for _, s := range []string{
+		"SET CLUSTER SETTING kv.rangefeed.enabled = true",
+		"SET CLUSTER SETTING kv.rangefeed.closed_timestamp_refresh_interval = '200ms'",
+		"SET CLUSTER SETTING kv.closed_timestamp.target_duration = '100ms'",
+		"SET CLUSTER SETTING kv.closed_timestamp.side_transport_interval = '50ms'",
+
+		"SET CLUSTER SETTING physical_replication.consumer.heartbeat_frequency = '1s'",
+		"SET CLUSTER SETTING physical_replication.consumer.job_checkpoint_frequency = '100ms'",
+		"SET CLUSTER SETTING physical_replication.consumer.minimum_flush_interval = '10ms'",
+		"SET CLUSTER SETTING physical_replication.consumer.cutover_signal_poll_interval = '100ms'",
+		"SET CLUSTER SETTING spanconfig.reconciliation_job.checkpoint_interval = '100ms'",
+	} {
+		sqlA.Exec(t, s)
+		sqlB.Exec(t, s)
+	}
+	compareAtTimetamp := func(ts string) {
+		fingerprintQueryFmt := "SELECT fingerprint FROM [SHOW EXPERIMENTAL_FINGERPRINTS FROM TENANT [%s]] AS OF SYSTEM TIME %s"
+		var fingerprintF int64
+		sqlA.QueryRow(t, fmt.Sprintf(fingerprintQueryFmt, "50", ts)).Scan(&fingerprintF)
+		var fingerprintG int64
+		sqlB.QueryRow(t, fmt.Sprintf(fingerprintQueryFmt, "51", ts)).Scan(&fingerprintG)
+		require.Equal(t, fingerprintF, fingerprintG, "fingerprint mismatch at %s", ts)
+	}
+
+	// The overall test plan looks like:
+	//
+	// SETUP
+	//   Create tenant 50 on serverA
+	//   Start service for tenant 50
+	//   Write to tenant 50
+	//   Replicate tenant 50 on serverA to tenant 51 on serverB
+	//
+	// FAILOVER
+	//   Complete replication on tenant 51 as of ts1
+	//   Fingerprint 50 and 51 as of ts1
+
+	// SETUP
+	t.Logf("creating tenant [50]")
+	sqlA.Exec(t, "CREATE VIRTUAL CLUSTER [50]")
+	sqlA.Exec(t, "ALTER VIRTUAL CLUSTER [50] START SERVICE SHARED")
+
+	tenFDB := newTenantConn(t, serverA.SystemLayer(), "cluster-50")
+	defer tenFDB.Close()
+	sqlTen50 := sqlutils.MakeSQLRunner(tenFDB)
+
+	sqlTen50.Exec(t, "CREATE DATABASE test")
+	sqlTen50.Exec(t, "CREATE TABLE test.t (k PRIMARY KEY) AS SELECT generate_series(1, 100)")
+
+	t.Logf("starting replication [50]->[51]")
+	sqlB.Exec(t, "CREATE VIRTUAL CLUSTER [51] FROM REPLICATION OF 'cluster-50' ON $1", serverAURL.String())
```

Comment:
This is the main line I'm interested in; I think the rest of the test is just verifying PCR works which you could probably skip since that should be covered in existing tests.

That said, I might suggest one more check after this line though that just verifies what is in system.tenants before and after this line, specifically that it actually made the tenant with id 51 and not something else.

Reply:
Done.